src/Provisioning/Provisioning.csproj

```diff
@@ -116,68 +116,191 @@
+    <Exec 
+        Command="npm list -g appium --json --depth 1 --loglevel error > &quot;$(NpmListJsonFile)&quot;" 
+        ContinueOnError="true"
+        StandardErrorImportance="high">
+        <Output TaskParameter="ExitCode" PropertyName="NpmListExitCode" />
+    </Exec>
+
+    <!-- Read the file content if npm list command was successful -->
+    <ReadLinesFromFile
+        File="$(NpmListJsonFile)"
+        Condition="Exists('$(NpmListJsonFile)') AND '$(NpmListExitCode)' == '0'">
+      <Output TaskParameter="Lines" ItemName="_AppiumNpmListJsonLines" />
+    </ReadLinesFromFile>
+
+    <!-- Concat the file lines into a single property -->
+    <PropertyGroup>
+      <_AppiumNpmListJson>@(_AppiumNpmListJsonLines, '')</_AppiumNpmListJson>
+    </PropertyGroup>
+
+    <!-- Peek the appium version info from the json output -->
+    <JsonPeek
+        Content="$(_AppiumNpmListJson)"
+        Query="$.dependencies.appium.version"
+        Condition="'$(_AppiumNpmListJson)' != ''">
 			<Output TaskParameter="Result" PropertyName="_AppiumNpmListVersion" />
 		</JsonPeek>
 
-		<!-- Install appium at the specified version if necessary -->
-		<Exec Command="npm i --location=global appium@$(AppiumVersion)" Condition=" '$(_AppiumNpmListVersion)' != '$(AppiumVersion)' " />
+    <!-- Log what we're going to do -->
+    <Message
+        Text="Installing Appium $(AppiumVersion) (current version: $(_AppiumNpmListVersion))"
+        Importance="high"
+        Condition="'$(_AppiumNpmListVersion)' != '$(AppiumVersion)'" />
+    <Message
+        Text="Appium is already at the required version $(AppiumVersion)"
+        Importance="normal"
+        Condition="'$(_AppiumNpmListVersion)' == '$(AppiumVersion)'" />
+
+    <!-- Install Appium at the specified version if necessary with timeout and privilege handling -->
+    <PropertyGroup Condition=" '$(IsMacOS)' == 'True' ">
+      <!-- macOS/Linux privilege elevation -->
+      <ElevatedNpmCommand>sudo -E npm i --location=global appium@$(AppiumVersion)</ElevatedNpmCommand>
+    </PropertyGroup>
+
+    <PropertyGroup Condition=" '$(IsMacOS)' != 'True' ">
+      <!-- Windows privilege elevation -->
+      <ElevatedNpmCommand>powershell -ExecutionPolicy Bypass -Command "Start-Process -FilePath 'cmd.exe' -ArgumentList '/c npm i --location=global appium@$(AppiumVersion)' -Verb RunAs -Wait"</ElevatedNpmCommand>
+    </PropertyGroup>
+
+    <!-- Try normal install first -->
+    <Exec
+        Command="npm i --location=global appium@$(AppiumVersion)"
+        Condition="'$(_AppiumNpmListVersion)' != '$(AppiumVersion)'"
+        Timeout="600000"
+        StandardOutputImportance="normal"
+        StandardErrorImportance="high"
+        ContinueOnError="true"
+        IgnoreExitCode="true">
+      <Output TaskParameter="ExitCode" PropertyName="NpmExitCode" />
+    </Exec>
+
+    <!-- If normal install fails, try with elevated privileges -->
+    <Exec
```

Comment:
This is one of the most important changes, retry install Appium drivers with elevated privileges.